src/ntp/client.c

```diff
@@ -618,22 +681,37 @@ static void *ntp_client_thread(void *arg)
 			restart_ftl("System time updated");
 		}
 
+		// Calculate time to sleep
+		unsigned int sleep_time = config.ntp.sync.interval.v.ui - (unsigned int)time_delta;
+
 		// Set first run to false
 		first_run = false;
 
-		if(success && !ntp_server_started)
+		if(!ntp_server_started)
 		{
-			// Initialize NTP server only after first NTP
-			// synchronization to ensure that the time is set
-			// correctly
-			ntp_server_started = ntp_server_start();
+			if(success)
+			{
+				// Initialize NTP server only after first high
+				// accuracy NTP synchronization to ensure that
+				// the time is set correctly
+				ntp_server_started = ntp_server_start();
+			}
+			else
+			{
+				log_debug(DEBUG_NTP, "Local time is too inaccurate, retrying before launching NTP server");
+
+				// Reduce retry time if the time is not accurate enough
+				if(retry_count++ < RETRY_INTERVAL &&
```

Comment:
```suggestion
				if(retry_count++ < 3 &&
```

`retry_count` is a small counter integer and will likely always be smaller than `RETRY_INTERVAL` which defaults to `600`.

Reply:
Oh, yes, copy-paste error, it should have been
```suggestion
				if(retry_count++ < RETRY_ATTEMPTS &&
```